src/app/dashboard/whitelist/actions.js

```diff
@@ -0,0 +1,281 @@
+//@flow
+
+import * as ui from 'polymath-ui'
+import { TransferManager, SecurityToken } from 'polymathjs'
+import type { Investor } from 'polymathjs/types'
+
+import { formName as userFormName } from './components/userForm'
+import { formName as editInvestorsFormName } from './components/editInvestorsForm'
+import type { GetState } from '../../../redux/reducer'
+import type { ExtractReturn } from '../../../redux/helpers'
+
+//ac_ = actionCreator
+export const TRANSFER_MANAGER = 'dashboard/whitelist/TRANSFER_MANAGER'
+export const ac_transferManager = (transferManager: TransferManager) => ({ type: TRANSFER_MANAGER, transferManager })
+
+export const UPLOAD_CSV = 'dashboard/whitelist/UPLOAD_CSV'
+export const ac_csvUpload = (csvMessage: string, addresses: Array<string>, sell: Array<string>, buy: Array<string>, previewCSVShowing: boolean, ) =>
+  ({ type: UPLOAD_CSV, csvMessage, addresses, sell, buy, previewCSVShowing })
+export const UPLOAD_CSV_FAILED = 'dashboard/whitelist/UPLOAD_CSV_FAILED'
+
+export const GET_WHITELIST = 'dashboard/whitelist/GET_WHITELIST'
+export const ac_getWhitelist = (investors: Array<Investor>) => ({ type: GET_WHITELIST, investors })
+// export const GET_WHITELIST_FAILED = 'dashboard/whitelist/GET_WHITELIST_FAILED'
+
+export const LIST_LENGTH = 'dashboard/whitelist/LIST_LENGTH'
+export const ac_listLength = (listLength: number) => ({ type: LIST_LENGTH, listLength })
+
+export type Action =
+  | ExtractReturn<typeof ac_transferManager>
+  | ExtractReturn<typeof ac_csvUpload>
+  | ExtractReturn<typeof ac_getWhitelist>
+  | ExtractReturn<typeof ac_listLength>
+
+//initialize grabs transferManager , and stores it in state for other functions to easily call
+//It then calls getWhiteList() to populate the table for the user
+export const initialize = () => async (dispatch: Function, getState: GetState) => {
+  const token = getState().token.token
+  if (!token || !token.contract) {
+    //eslint-disable-next-line
+    console.error('Contract manager object not found, it did not carry over into the state')
+    return
+  }
+  const contract: SecurityToken = token.contract
+  const transferManager: TransferManager = await contract.getTransferManager()
+  dispatch(ac_transferManager(transferManager))
+  dispatch(getWhitelist())
+}
+
+//Uploads the CSV file, reads it with built in js FileReader(), dispatches to the store the csv file information,
+//which can then be sent to the blockchain with multiUserSumbit()
+//QUESTION: @davekaj - Do we need to limit CSV file to 50 or 100, and notify them that it is too long? also keep in mind gas limit and WS packet size
+export const uploadCSV = (file: Object) => async (dispatch: Function) => {
```

Comment:
probably you can use File interface here as a type instead of Object https://developer.mozilla.org/en-US/docs/Web/API/File

Reply:
fixed